main.go

```diff
@@ -1,167 +1,97 @@
+	"github.com/sensu/sensu-plugins-go-library/sensu"
 	"strings"
 
 	"github.com/bluele/slack"
-	"github.com/sensu/sensu-go/types"
-	"github.com/spf13/cobra"
 )
 
-type HandlerConfigOption struct {
-	Value string
-	Path  string
-	Env   string
-}
-
 type HandlerConfig struct {
-	SlackWebhookUrl HandlerConfigOption
-	SlackChannel    HandlerConfigOption
-	SlackUsername   HandlerConfigOption
-	SlackIconUrl    HandlerConfigOption
-	Timeout         int
-	Keyspace        string
+	sensu.PluginConfig
+	SlackWebhookUrl string
+	SlackChannel    string
+	SlackUsername   string
+	SlackIconUrl    string
 }
 
 var (
-	stdin  *os.File
 	config = HandlerConfig{
-		// default values
-		SlackWebhookUrl: HandlerConfigOption{Path: "webhook-url", Env: "SENSU_SLACK_WEHBOOK_URL"},
-		SlackChannel:    HandlerConfigOption{Path: "channel", Env: "SENSU_SLACK_CHANNEL"},
-		SlackUsername:   HandlerConfigOption{Path: "username", Env: "SENSU_SLACK_USERNAME"},
-		SlackIconUrl:    HandlerConfigOption{Path: "icon-url", Env: "SENSU_SLACK_ICON_URL"},
-		Timeout:         10,
-		Keyspace:        "sensu.io/plugins/slack/config",
+		PluginConfig: sensu.PluginConfig{
+			Name:     "sensu-slack-handler",
+			Short:    "The Sensu Go Slack handler for notifying a channel",
+			Timeout:  10,
+			Keyspace: "sensu.io/plugins/slack/config",
+		},
 	}
-	options = []*HandlerConfigOption{
-		// iterable slice of user-overridable configuration options
-		&config.SlackWebhookUrl,
-		&config.SlackChannel,
-		&config.SlackUsername,
-		&config.SlackIconUrl,
+
+	slackConfigOptions = []*sensu.PluginConfigOption{
+		{
+			Path:      "webhook-url",
+			Env:       "SENSU_SLACK_WEHBOOK_URL",
+			Argument:  "webhook-url",
+			Shorthand: "w",
+			Default:   "",
+			Usage:     "The webhook url to send messages to, defaults to value of SLACK_WEBHOOK_URL env variable",
+			Value:     &config.SlackWebhookUrl,
+		},
+		{
+			Path:      "channel",
+			Env:       "SENSU_SLACK_CHANNEL",
+			Argument:  "channel",
+			Shorthand: "c",
+			Default:   "#general",
+			Usage:     "The channel to post messages to",
+			Value:     &config.SlackChannel,
+		},
+		{
+			Path:      "username",
+			Env:       "SENSU_SLACK_USERNAME",
+			Argument:  "username",
+			Shorthand: "u",
+			Default:   "sensu",
+			Usage:     "The username that messages will be sent as",
+			Value:     &config.SlackUsername,
+		},
+		{
+			Path:      "icon-url",
+			Env:       "SENSU_SLACK_ICON_URL",
+			Argument:  "icon-url",
+			Shorthand: "i",
+			Default:   "http://s3-us-west-2.amazonaws.com/sensuapp.org/sensu.png",
+			Usage:     "A URL to an image to use as the user avatar",
+			Value:     &config.SlackIconUrl,
+		},
 	}
 )
 
 func main() {
-	rootCmd := configureRootCommand()
-	if err := rootCmd.Execute(); err != nil {
-		log.Fatal(err)
+	goHandler, _ := sensu.NewGoHandler(&config.PluginConfig, slackConfigOptions, checkArgs, executeHandler)
+	err := goHandler.Execute()
+	if err != nil {
+		fmt.Printf("Error executing plugin: %s", err)
```

Comment:
Could you explain the reasoning for the change from `log.Fatal` to `fmt.Printf`?

Reply:
After upgrading to use "0.2.0" version of "sensu-plugins-go-library" library, the printing of error is now being handled by the library itself. It simply logs the error to stderr and then exits with status 1.